sphynx/networkit/networkit.swigcxx

```diff
@@ -0,0 +1,23 @@
+%module networkit
+
+%{
+/*
+#include "example.h"
+*/
+#include <networkit/centrality/Betweenness.hpp>
+#include <networkit/generators/BarabasiAlbertGenerator.hpp>
+using namespace NetworKit;
+%}
```

Comment:
Sorry, I'm a total SWIG noob, but probably it's worth for me to understand a bit what's going on here for the sake of future PRs. So some questions:
- What's the deal with the stuff enclosed in `%{}%`? Why does it "duplicate" in a sense what's included below?
- What's with the apparently commented out `example.h` includes?
- Most importantly, what's the effect of this file? We ask swig to create wrappers for everything that we `%include`? Or that we `#inlcude`? Or something else? :)

Reply:
> Sorry, I'm a total SWIG noob

That makes two of us. 😄 

> * What's the deal with the stuff enclosed in `%{}%`? Why does it "duplicate" in a sense what's included below?

A C++ language wrapper will be generated from this file. The stuff in `%{}%` will be put into the C++ code verbatim. It's for the C++ compiler.

The other part defines the interface that SWIG will generate wrappers for. The language for defining the interface is _almost_ C++. So we can `%include` unaltered C++ headers in many cases.

There are just two tricks here: `#define final` because the SWIG parser does not know about the `final` keyword that NetworKit classes use. And `class Graph` because SWIG couldn't parse `Graph.hpp` and I also didn't need to wrap it for this test. I'll sure need it later, but I can write a simpler version to define a limited interface for SWIG to wrap. I don't mind if I have to do that for a few base classes as long as wrapping the operations is cheap.

> * What's with the apparently commented out `example.h` includes?

I forgot to delete them. 😓

Reply:
Thanks a lot, this helps! One more thing. So what is a C++ wrapper in this context? Here is my theory, tell me whether this is accurate...

So, it is still C++ code, defining C++ stuff (functions? classes?). These stuff use the original C++ stuff for their implementation. The reason we need them, and can't use the original "stuff" is that go is limited in what kind of stuff it can directly call. So we are wrapping non-go-compliant kind of stuff within go-compliant kind of stuff.

Then, there must be one more thing being generated here: go code which implements or declares the calls to the "C++ wrapper", the go-compliant stuff from above.

Reply:
> Thanks a lot, this helps!

I'm also adding a few comments to explain this in the file.

> One more thing. So what is a C++ wrapper in this context? Here is my theory, tell me whether this is accurate...
> 
> So, it is still C++ code, defining C++ stuff (functions? classes?). These stuff use the original C++ stuff for their implementation. The reason we need them, and can't use the original "stuff" is that go is limited in what kind of stuff it can directly call. So we are wrapping non-go-compliant kind of stuff within go-compliant kind of stuff.
> 
> Then, there must be one more thing being generated here: go code which implements or declares the calls to the "C++ wrapper", the go-compliant stuff from above.

That's exactly how it seems to work. It generates a bunch of files like `networkit_wrap.cxx` and `_networkit_swig.cgo1.go`. Go cannot straight up access C++, so I think the C++ side of the wrapper creates a C interface. (And does who knows what else. I haven't looked.)

There are two things I like:
- This is all handled by the Go tooling. I just run `go build` and when it sees the `.swigcxx` file it runs a bunch of commands to take care of everything.
- SWIG has no problem automatically wrapping the algorithms. Once the basics are set up I think it will be easy to access the whole library.

I've got the Sphynx &rarr; NetworKit direction working for passing in a graph now. Unfortunately the NetworKit `Graph` object is much more complex than our `EdgeBundle`, so this is far from zero-copy. At least it doesn't go through disk or Python. 😄